x/programs/cmd/simulator/vm/actions/program_execute.go

```diff
@@ -107,8 +106,12 @@ func (t *ProgramExecute) Execute(
 	importsBuilder.Register("state", func() host.Import {
 		return pstate.New(logging.NoLog{}, mu)
 	})
+	callContext := program.Context{
+		ProgramID:        programID,
+		Actor:            [32]byte(actor[1:]),
```

Comment:
I recommend passing all 33 bytes here as there could be conflicting addresses within the trailing 32